### PR TITLE
[HS-1112780] Force sign-in after 422 response in designation editor

### PR DESCRIPTION
**Pull request by canac**
Would this fix the issue? If loading the content responds with a 422, we ask the user to sign in again and try again to load the content.

I wasn't able to test in staging because of conflicts with Daniel's Okta work. But I did test it locally.

https://secure.helpscout.net/conversation/2525923620/1112780?viewId=7296729

### CODE REVIEW

File: src/app/designationEditor/designationEditor.spec.js [hunk 1]
```diff
@@ -176,6 +176,24 @@ describe('Designation Editor', function () {
...
+      $httpBackend.expectGET(designationConstants.designationEndpoint + '?designationNumber=' + designationSecurityResponse.designationNumber)
+        .respond(422, null)
+      $httpBackend.expectGET(designationConstants.designationImagesEndpoint + '?designationNumber=' + designationSecurityResponse.designationNumber)
+        .respond(422, null)
+
+      $ctrl.getDesignationContent().then(() => {
+        expect($ctrl.sessionModalService.open).toHaveBeenCalled()
+        expect($ctrl.designationContent).toBeDefined()
+        expect($ctrl.contentLoaded).toEqual(true)
+        expect($ctrl.loadingContentError).toEqual(false)
```
Comment (wrandall22) [inline src/app/designationEditor/designationEditor.spec.js:192]: I wonder if these expectations are being called or if we need to add `done()` to this test since it is asynchronous. Did you test to find out for sure if these are being called?

**Comment by wrandall22** [comment]
This certainly could fix the issue. The edge case in my mind is what happens if the login doesn't properly fix the cookies and they keep getting 422 responses. I suppose we can fix that bridge if it comes up.

**Comment by canac** [comment]
> The edge case in my mind is what happens if the login doesn't properly fix the cookies and they keep getting 422 responses.

Maybe I should add logic to make sure that the login+retry only happens once.

File: src/app/designationEditor/designationEditor.component.js [hunk 2]
```diff
@@ -93,17 +95,28 @@ class DesignationEditorController {
       // get designation photos
       this.designationEditorService.getPhotos(this.designationNumber)
     ]).then(responses => {
-      this.contentLoaded = true
```
Comment (canac) [inline src/app/designationEditor/designationEditor.component.js:96]: I removed this because `this.updateCarousel` synchronously sets `this.contentLoaded` to `false`. It made tests confusing because in the test you would expect `this.contentLoaded === true` unless you check the body of `this.updateCarousel`. But maybe it would be better to remove the `this.updateCarousel` here and rely on the fact that `this.contentLoaded` is changed in this `.then()` handler to refresh the carousels. @wrandall22 Do you have thoughts on which approach is clearer?

Comment (wrandall22) [reply src/app/designationEditor/designationEditor.component.js:96]: I think calling the function is clearer to me

**Comment by canac** [comment]
@wrandall22 Do you want to give this another review?

**Comment by canac** [comment]
@wrandall22 When I tried it locally, a successful login does not trigger a page refresh.
